### PR TITLE
Add total time logging at the end of the simulation

### LINKED ISSUE #1907: Display total simulation time at the end of the simulation
Format = XXhYYmZZs ?

### PR DESCRIPTION
**Pull request by payetvin**
close  #1907

### CODE REVIEW

**Review by a-zakir — APPROVED** [review]

**Comment by a-zakir** [comment]
sonar complains about sprintf...

File: src/solver/application/application.cpp [hunk 1]
```diff
@@ -428,6 +428,20 @@ void Application::writeComment(Data::Study& study)
...
 
+static void logTotalTime(unsigned duration)
+{
+    duration /= 1000; // duration in seconds instead of ms
+    unsigned hours = duration / 3600;
+    unsigned minutes = duration / 60 % 60;
+    unsigned seconds = duration % 60;
+
+    char timeDisplay[32];
+    sprintf(timeDisplay, "%02uh%02um%02us", hours, minutes, seconds);
```
Comment (flomnes) [inline src/solver/application/application.cpp]: Use snprintf to avoid possible overflow

File: src/solver/application/application.cpp [hunk 2]
```diff
@@ -428,6 +428,20 @@ void Application::writeComment(Data::Study& study)
     }
 }
 
+static void logTotalTime(unsigned duration)
+{
+    duration /= 1000; // duration in seconds instead of ms
+    unsigned hours = duration / 3600;
```
Comment (flomnes) [inline src/solver/application/application.cpp]: Consider using std::chrono, do not re-invent the wheel

**Review by flomnes — CHANGES_REQUESTED** [review]

File: src/solver/application/application.cpp [hunk 3]
```diff
@@ -428,6 +428,22 @@ void Application::writeComment(Data::Study& study)
...
+{
+    std::chrono::milliseconds d(duration);
+    auto hours = std::chrono::duration_cast<std::chrono::hours>(d);
+    d -= hours;
+    auto minutes = std::chrono::duration_cast<std::chrono::minutes>(d);
+    d -= minutes;
+    auto seconds = std::chrono::duration_cast<std::chrono::seconds>(d);
+
+    char timeDisplay[32];
+    snprintf(timeDisplay, 10, "%02luh%02lum%02lus", hours.count(), minutes.count(), seconds.count());
```
Comment (flomnes) [inline src/solver/application/application.cpp]: Actually, you don't even need to bother with `snprintf`, you can use

```cpp
logs.info().appendFormat(format, args);
```

See for example **src/solver/optimisation/opt_appel_solveur_quadratique.cpp**
```cpp
logs.info().appendFormat("Variable %ld cout lineaire %e  cout quadratique %e",
                         i,
                         ProblemeAResoudre->CoutLineaire[i],
                         ProblemeAResoudre->CoutQuadratique[i]);
```

**Review by flomnes — APPROVED** [review]

**Pull request merged**